openedx/core/djangolib/js_utils.py

```diff
@@ -69,6 +69,13 @@ def dump_js_escaped_json(obj, cls=EdxJSONEncoder):
         (string) Escaped encoded JSON.
 
     """
+    if isinstance(obj, dict) and 'courseId' in obj:
```

Comment:
Why you made these changes ? Please add some explanation here.

Reply:
in some instances `courseId` and `usageId` would have a `bytes` value that can't be used in `json.dumps`, so for that case we're doing `decode`